python/mxnet/optimizer.py

```diff
@@ -299,8 +300,13 @@ def update(self, index, weight, grad, state):
         if state:
             mom = state
             mom[:] *= self.momentum
-            mom[:] += -lr * (grad + wd * weight)
-            weight[:] += mom
+            grad += wd * weight
```

Comment:
What problem can this cause? In the previous line we've already done this:
`grad = grad * self.rescale_grad`